osquery/config/config.cpp

```diff
@@ -86,7 +87,23 @@ void Config::clearFiles() {
   files_.erase(files_.begin(), files_.end());
 }
 
+bool Config::isValid() {
+  ReadLock rlock(config_valid_mutex_);
+  return valid_;
+}
+
+void Config::setValid() {
+  WriteLock wlock(config_valid_mutex_);
+  valid_ = true;
+}
+
+void Config::setInvalid() {
+  WriteLock wlock(config_valid_mutex_);
```

Comment:
`Config::load()` will definitely one day be called from another thread, but a literal `bool` wont tear so there's no need to read/write lock around it. If you're worried about core cache synchronization then you can wrap it in a `std::atomic<bool>`. But that's a bit overkill since the reading will happen in a super-lazy state.